pkg/build/strategies/onbuild/onbuild.go

```diff
@@ -90,19 +90,15 @@ func (builder *OnBuild) Build(config *api.Config) (*api.Result, error) {
 	// Change the installation directory for this config to store scripts inside
 	// the application root directory.
 	if err := builder.source.Prepare(config); err != nil {
-		return buildResult, err
+		return nil, err
 	}
 
 	// If necessary, copy the STI scripts into application root directory
 	builder.copySTIScripts(config)
 
 	glog.V(2).Info("Creating application Dockerfile")
 	if err := builder.CreateDockerfile(config); err != nil {
-		buildResult.BuildInfo.FailureReason = utilstatus.NewFailureReason(
-			utilstatus.ReasonDockerfileCreateFailed,
-			utilstatus.ReasonMessageDockerfileCreateFailed,
-		)
-		return buildResult, err
```

Comment:
not sure why you say it "doesn't work".  Whether the caller currently respects the return value or not, the buildResult object is correct/valid, no?

I'd rather be returning it so that if we ever turn on-build back on for openshift, we don't forget to reintroduce this code.